charts/gha-runner-scale-set/values.yaml

```diff
@@ -44,6 +44,36 @@ githubConfigSecret:
 
 # runnerGroup: "default"
 
+## Custom CA certificates can be provided by bundling them in a configmap and
+## referring it in githubServerTLS.certConfigMapRef, for example:
+#
+# githubServerTLS:
+#   certConfigMapRef: your-configmap-name
+#
+## If you do need custom CA certificates do remember that the runner (and your
```

Comment:
should we do this under `ephemeral_runner`, just like we do it for proxy?

Reply:
I think there's an argument for leaving this to the pod spec, how and where you want to mount certificates in the container changes with the image being used. And depending on the image and actions being used users may need to set specific env vars too?

Reply:
yeah this is going to be very tricky for runners. I think this should be something that end users have to deal with on a case by case basis. Trust stores vary greatly from one distribution to another

Reply:
I still think we might want to do some work to get the runner to work with the self-sign cert out of the box, similar to the proxy setting. For advance users, they can always do whatever they want. I could be wrong, btw. 😄

Reply:
I think tackling our own runner image is a good compromise while adding documentation for how it can be done for other images

Reply:
Even with our own runner image mounting the configmap is only part of the picture, specific actions can require other customisations from the user. Personally I think documentation can go a long way, other than that what we can do is if `image == ghcr.io/actions/actions-runner` we mount certificates under `/usr/local/share/ca-certificates/`?

Reply:
And set `NODE_EXTRA_CA_CERTS` env var to cover first party node actions?

Reply:
`NODE_EXTRA_CA_CERTS` [takes a single file](https://nodejs.org/api/cli.html#node_extra_ca_certsfile), and by using configmaps we could potentially have multiple. Should we change the spec to point to a [configmap key](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#configmapkeyselector-v1-core) instead?

Reply:
Or we can have exported constant that we document saying: this key is special to us. Basically, similar to the runner being a special name for the EphemeralRunner, or `dind` being a special name.

Reply:
> other than that what we can do is if image == ghcr.io/actions/actions-runner we mount certificates under /usr/local/share/ca-certificates/

Maybe, always mount if the field is populated without checking the image? Because what if rename the image or publish somewhere else or ... 

Also, I think just adding the files will not do, we need to run `update-ca-certificates` at some point right?

Reply:
Node... *sigh*

Reply:
We've decided to change the spec for `githubServerTLS` to take a direct reference to a configmap key (leaves the door open for other methods down the line) and a `runnerMountPath` defaulting to `/usr/local/share/ca-certificates/` if not specified. When creating runner pods we'll respect that and set `NODE_EXTRA_CA_CERTS` to point to the same path.